### ISSUE TITLE
Duplicate `itoa` 0.4 via `csv` dev-dependency

### ISSUE THREAD

**Issue by pinkforest** [dalek-cryptography/curve25519-dalek#430]
`csv` has not had a release for a year and it's stuck at 0.4 itoa that uses unsound `std::mem::uninitialized`

https://github.com/BurntSushi/rust-csv/issues/271

This causes `itoa` dependency duplication

This is brought by `criterion` dev-dependency
```
 cargo tree -i csv
csv v1.1.6
└── criterion v0.3.6
    [dev-dependencies]
    └── curve25519-dalek v4.0.0-pre.2 (/home/foobar/pp/curve25519-dalek)
```

Currently `criterion` v0.3.6 is used and v0.4.0 is the latest

Looking at the `criterion` crate the `csv` dependency is optional in 0.4.0
`csv` crate used to be non-optional and default dependency in 0.3.6

So bumping up `criterion` will solve this.

```
warning[B004]: found 2 duplicate entries for crate 'itoa'
   ┌─ /home/foobar/pp/curve25519-dalek/Cargo.lock:30:1
   │  
30 │ ╭ itoa 0.4.8 registry+https://github.com/rust-lang/crates.io-index
31 │ │ itoa 1.0.4 registry+https://github.com/rust-lang/crates.io-index
   │ ╰────────────────────────────────────────────────────────────────^ lock entries
   │  
   = itoa v0.4.8
     └── csv v1.1.6
         └── criterion v0.3.6
             └── (dev) curve25519-dalek v4.0.0-pre.2
   = itoa v1.0.4
     └── serde_json v1.0.87
         ├── criterion v0.3.6
         │   └── (dev) curve25519-dalek v4.0.0-pre.2
         └── tinytemplate v1.2.1
             └── criterion v0.3.6 (*)
```

**Comment by pinkforest** [comment]
Ok first stab at `default-features = false`

Needs plotters:

Gnuplot not found, using plotters backend
thread 'main' panicked at 'Criterion was built without plotters support.', /home/foobar/.cargo/registry/src/github.com-1ecc6299db9ec823/criterion-0.4.0/src/lib.rs:302:42

criterion:

```toml
[features]
stable = [
  "csv_output",
  "html_reports",
  "async_futures",
  "async_smol",
  "async_tokio",
  "async_std",
]
default = ["rayon", "plotters", "cargo_bench_support"]
```

ok so enabling plotters and see again:
$ `cargo +nightly tarpaulin --run-types Tests,Doctests,Benchmarks,Examples,Lib,Bins`

**Comment by pinkforest** [comment]
Success :partying_face: 
```

Nov 11 23:33:11.669  INFO cargo_tarpaulin::report: Coverage Results:
|| Uncovered Lines:
|| benches/dalek_benchmarks.rs: 95-96, 103, 106, 125, 128, 149-150, 152, 180-181, 208-209, 266, 268, 271, 326, 328-329
|| src/backend/serial/curve_models/mod.rs: 189-192, 210-214, 246-247, 291, 293-296, 309, 311-313, 525-526, 529, 535-536, 539, 545-546, 549, 555-557
|| src/backend/serial/fiat_u32/field.rs: 72-76, 82-87, 92-96, 102-107, 112-114, 120-123, 129-134
|| src/backend/serial/fiat_u64/field.rs: 61-65, 71-76, 81-85, 91-96, 101-103, 109-112, 118-123, 183-186
|| src/backend/serial/scalar_mul/pippenger.rs: 150
|| src/backend/serial/scalar_mul/straus.rs: 142
|| src/backend/serial/u32/field.rs: 68-70, 77-80, 85, 89-100, 102, 108-111, 116-118, 126, 129-131, 135-136, 152, 156-157, 182-184, 186-195, 224, 230-233, 345-347, 349-350, 353-354, 408-409, 413-414, 570-571
|| src/backend/serial/u32/scalar.rs: 51-52, 212-213, 241-245, 248-252, 264, 270-279, 283-299, 306, 308-311, 314-317, 321, 324-332, 335-343, 346, 351-353, 359-361, 366-367, 372-373, 378-379
|| src/backend/serial/u64/field.rs: 45-46, 51-52, 69, 74-76, 103-105, 116, 219, 238-243, 292, 462
|| src/backend/serial/u64/scalar.rs: 27-28, 53, 200, 213, 219, 243, 245-246, 251-252, 281-282, 289-290, 296, 302
|| src/backend/vector/avx2/edwards.rs: 235-237, 245, 252, 254, 258, 261, 264, 267, 269, 274, 286-287, 292-296, 298, 303-308, 311, 316-321, 324
|| src/backend/vector/avx2/field.rs: 61-64, 68-69, 71, 84, 92-93, 98, 211-213, 217-227, 232, 236-241, 251-253, 277-279, 281-282, 284-285, 288-289, 292-293, 296-297, 300-301, 305-307, 314-319, 377, 381-386, 400, 402, 404, 406, 415-416, 418-425, 434, 439-440, 458, 461, 465, 467-468, 470-471, 473-474, 476-477, 479-480, 491, 497, 501, 505, 514, 524, 526-527, 530-532, 534-535, 538-539, 544-547, 550, 561-562, 568, 584-589, 605, 607, 611, 613, 708-714, 724-730, 743, 747, 749-753, 755-765, 786-788, 790, 793-794, 796, 799-803, 805-809, 811, 816-821, 825-827, 829-838, 878
|| src/backend/vector/ifma/edwards.rs: 129-130, 132, 135, 138, 141, 144, 146, 150, 179-181, 189-190, 195-199, 201, 206-211, 214, 219-224, 227
|| src/backend/vector/ifma/field.rs: 21, 23, 28, 30, 56, 60-70, 86, 90-96, 156, 158, 160, 162, 166-174, 179-185, 190-196, 214-222, 227-234, 240-246, 251-257, 262, 264, 267-275, 278-284, 286-287, 289-291, 293-296, 298-302, 304-309, 311-316, 318-322, 324-327, 329-331, 333-334, 336-339, 341-343, 345-346, 348-351, 353-357, 359-363, 365-370, 378-379, 385-387, 390-394, 397-402, 411-417, 425, 427, 429-432, 434, 436-445, 448-455, 458-460, 462-467, 476, 479-480, 483-491, 494-503, 511-518, 521-528, 531-538, 540-541, 544-551, 553-554, 557-564, 566, 569-576, 578-580, 583-590, 593-600, 603-604, 606-611
|| src/backend/vector/scalar_mul/pippenger.rs: 37-42, 44, 47-48, 53, 55, 57, 59-61, 66-67, 70, 72-73, 80, 82-88, 100-105, 108, 113, 116-118, 120
|| src/backend/vector/scalar_mul/precomputed_straus.rs: 39, 59, 61, 63, 65, 68, 70, 73-76, 81-83, 85-90, 94-99, 104
|| src/backend/vector/scalar_mul/straus.rs: 49, 51, 54, 56, 59, 61-65, 67, 70, 83, 85, 87, 89, 92, 94-95, 97-101, 106
|| src/edwards.rs: 166-167, 172-173, 232-234, 236, 247-249, 251, 263-264, 266-267, 270, 274-278, 280, 282, 286, 298-299, 301-302, 305, 309-313, 315, 319, 351-352, 362-363, 365, 367, 383-384, 394-396, 402-406, 613-614, 803, 850-851, 956-959, 961, 985-986, 991-992, 1094-1095, 1098
|| src/lib.rs: 247-249, 262, 267, 275, 281, 284, 288-289, 294
|| src/macros.rs: 9-10, 26-27, 42, 56-57, 63-64, 81-82
|| src/montgomery.rs: 84-85, 105-106, 112-113, 217-218, 347-348
|| src/ristretto.rs: 219-220, 226-227, 240-241, 243, 245, 306, 325-326, 350-352, 354, 365-367, 369, 381-382, 384-385, 388, 392-396, 398, 400, 404, 416-417, 419-420, 423, 427-431, 433, 437, 537, 543, 564, 774-775, 827-828, 837-838, 849-850, 885-887, 994, 1001, 1006, 1045-1046, 1050-1051, 1098-1099, 1104-1106, 1119-1120, 1125-1126
|| src/scalar.rs: 265-266, 327-328, 336-337, 352, 354, 367, 380-383, 401-403, 405, 417-418, 420-421, 424, 428-432, 434-435, 440, 469-470, 540-541, 971, 975, 1004-1005, 1010
|| src/traits.rs: 342
|| src/window.rs: 83-84, 86-87, 90, 153-154, 171-175, 178, 196-199, 201, 206-210, 213
|| Tested/Total Lines:
|| benches/dalek_benchmarks.rs: 140/159
|| src/backend/serial/curve_models/mod.rs: 111/143
|| src/backend/serial/fiat_u32/field.rs: 0/35
|| src/backend/serial/fiat_u64/field.rs: 0/39
|| src/backend/serial/scalar_mul/pippenger.rs: 37/38
|| src/backend/serial/scalar_mul/precomputed_straus.rs: 30/30
|| src/backend/serial/scalar_mul/straus.rs: 28/29
|| src/backend/serial/scalar_mul/variable_base.rs: 17/17
|| src/backend/serial/scalar_mul/vartime_double_base.rs: 24/24
|| src/backend/serial/u32/field.rs: 0/68
|| src/backend/serial/u32/scalar.rs: 0/83
|| src/backend/serial/u64/field.rs: 199/220
|| src/backend/serial/u64/scalar.rs: 148/165
|| src/backend/vector/avx2/edwards.rs: 0/34
|| src/backend/vector/avx2/field.rs: 0/199
|| src/backend/vector/ifma/edwards.rs: 0/34
|| src/backend/vector/ifma/field.rs: 0/328
|| src/backend/vector/scalar_mul/pippenger.rs: 0/40
|| src/backend/vector/scalar_mul/precomputed_straus.rs: 0/27
|| src/backend/vector/scalar_mul/straus.rs: 0/25
|| src/constants.rs: 3/3
|| src/edwards.rs: 162/232
|| src/field.rs: 72/72
|| src/lib.rs: 10/21
|| src/macros.rs: 16/27
|| src/montgomery.rs: 80/90
|| src/ristretto.rs: 197/271
|| src/scalar.rs: 248/287
|| src/traits.rs: 19/20
|| src/window.rs: 48/72
|| 
56.11% coverage, 1589/2832 lines covered
```

**Comment by rozbb** [comment]
Closed by #432
